### ISSUE TITLE
Create v0.10.0.dev tag

### ISSUE THREAD

**Issue by bashtage** [statsmodels/statsmodels#5186]
Need a dev tag in statsmodels/statsmodels to get correct version numbers now that we switched to versioneer.

Ideally 

```
git tag -a v0.10.0.dev0 ff0d659fdcaf746e0472f2c4b4c348c8a11b92d2
```
which corresponds to the same commit where the original setup was modified to have version 0.10.0dev,

https://github.com/statsmodels/statsmodels/commit/ff0d659fdcaf746e0472f2c4b4c348c8a11b92d2#diff-2eeaed663bd0d25b7e608891384b7298

**Comment by josef-pkt** [comment]
on command line (It's the first time I create a tag without github web interface)

```
>git tag -a v0.10.0.dev0 ff0d659fdcaf746e0472f2c4b4c348c8a11b92d2
<write tag message in text editor>
>git push origin v0.10.0.dev0
```

@bashtage  The tag is there.
How do we see that versioneer picks it up?
Do we have to do this after every release when we increment the version number to the next `dev`?

**Comment by bashtage** [comment]
It should show up in Travis builds soon.

The release procedure should look like:

1. Fork v.NEXT branch
2. On fork, make any final cleanups.  When done, tag v.NEXT and release (this can be done thorugh github).  You must release with 0 modifications after the tag. 
3. Make an empty commit on master, and then tag this commit as v.NEXT+1.dev0.  The prevents the v.NEXT+1.dev0 from appearing in the v.NEXT release branch.

**Comment by bashtage** [comment]
I can see the tag on Github

**Comment by thequackdaddy** [comment]
@bashtage @josef-pkt , release notes are here 

https://www.statsmodels.org/stable/dev/maintainer_notes.html#releasing

I think they are pretty good. 

One larger comment from the docs is that the way we laid out the site is probably a little sub-ideal. The table of contents is on the bottom, not on the side. The sidebar is bookmarks on the current page. I think a better idea would be a TOC on the sidebar, then on the top of the page the bookmarks for the current page.

**Comment by bashtage** [comment]
@thequackdaddy I think the site could use a big refresh with a better layout and ideally a more modern aesthetic.  This isn't my comparative advantage.

**Comment by bashtage** [comment]
http://scikit-learn.org/stable/ is decent for content-filled page. The best among numpy, scipy, sm, and pandas IMO.

**Comment by bashtage** [comment]
https://www.tensorflow.org/ is great, but I wouldn't expect anything different given its lineage.

**Comment by josef-pkt** [comment]
We lost the top of the website in the sourceforge to github doc transition.
Now it's essentially just a plain sphinx doc site with a few extras.

The problem always was that it is written and "designed" by applied econometricians or contributors in similar applied fields that figured out some css and java script.
It was never my comparative advantage either, except for reviewing a bit the content structure.

two non-Python for comparison

R: 
plain open source versus commercial open source
https://www.r-project.org/
https://mran.microsoft.com/

Stata which I look at whenever I get a new release announcement
https://www.stata.com/
https://www.stata.com/features/documentation/

(Aside to evangelizing statsmodels, more related to a comment in a different issue: 
Most of the time I think of statsmodels as a "sleeper" in movie terms. That is. there is no big advertising budget but eventually everyone will be using it. because it's the best and it's in Python. 
Corollary: We don't oversell statsmodels just to attract users. We are on the coat tails of the success of Python. Pandas and to some extent scikit-learn and we got a "captured" audience when users realize that they need some statistics and econometrics.
I'm thinking about this when I read some discussion threads in the Julia forums.

However, currently we are underselling, and leave the advertising mostly to blogpost by users and university and online courses.)

**Comment by bashtage** [comment]
The release process is in the docs.
